drivers/include/periph/compat.h

```diff
@@ -0,0 +1,157 @@
+/*
```

Comment:
The idea was to use the same file to also include `spi_t`, `uart_t` and so on until they are optimized for a certain platform instead of introducing one file for each module...